src/scripting/lua_root.cc

```diff
@@ -492,6 +493,26 @@ int LuaDescriptions::get_immovable_descriptions(lua_State* L) {
 	return 1;
 }
 
+/* RST
+   .. attribute:: building_descriptions
+
+      .. versionadded:: 1.2
+
+      (RO) An :class:`array` of all :class:`~wl.map.BuildingDescription` objects that are available.
+*/
+int LuaDescriptions::get_building_descriptions(lua_State* L) {
+	const Widelands::Descriptions& descriptions = get_egbase(L).descriptions();
+	lua_newtable(L);
+	int index = 1;
+	for (Widelands::DescriptionIndex i = 0; i < descriptions.nr_buildings(); ++i) {
+		lua_pushint32(L, index++);
```

Comment:
No need for variable `index`, you can just push `i + 1` here

Reply:
The variable `index` is in many (or all?) `get_..._descriptions()` functions :astonished: